src/napari_graph/base_graph.py

```diff
@@ -245,6 +247,9 @@ def __init__(
                     f"`edges` (shape: {edges.shape}) must have shape E x 2."
                 )
 
+            if self._coords is None:
+                self.add_nodes(np.unique(edges))
```

Comment:
Maybe move this to the "initialize edges" part (finish the validation first)? Also, what happens when the nodes are present already (because `coords` were given)?